libimage/manifest_list.go

```diff
@@ -210,8 +217,8 @@ func (i *Image) IsManifestList(ctx context.Context) (bool, error) {
 }
 
 // Inspect returns a dockerized version of the manifest list.
-func (m *ManifestList) Inspect() (*manifest.Schema2List, error) {
-	return m.list.Docker(), nil
+func (m *ManifestList) Inspect() (ManifestInspectReport, error) {
```

Comment:
@vrothberg I maybe wrong but it looks like hard to maintain to manually copy fields from one type to another. I think in `buildah inspect` we generate output for different formats in entirely different fields https://github.com/containers/buildah/blob/main/buildah.go#L137

Reply:
"Hard to maintain" is no good excuse for breaking users :^)  I prefer going the extra mile now and keep users happy.

I don't expect it to be difficult to maintain as the Docker spec is frozen and the OCI one is pretty much complete.  `image inspect` follows a similar approach and I don't recall it causing much maintenance burden w.r.t. adding new fields.

Reply:
@vrothberg Ah i see suggested approach SGTM afaics we will use this new type only to generate inspect output. Any objections if we maintain this new type in `libimage/manifest_list.go` so we keep it close to `inspect` code ?

Reply:
SGTM we are free to change the API of libimage.  We just need to update podman and buildah rather quickly to prevent other work from being blocked by the build failures.